src/main/java/de/cas/mse/exercise/csv/CsvReader.java

```diff
@@ -1,19 +1,53 @@
 package de.cas.mse.exercise.csv;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import de.cas.mse.exercise.csv.ui.CsvUi;
+import de.cas.mse.exercise.csv.ui.Filterable;
 
-public class CsvReader {
+public class CsvReader implements Filterable {
 	
-	private CsvUi csvUi;
+    private CsvUi csvUi;
+    private CsvFile csvFile;
 
+    // should be renamed: read()
 	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
-	}
+        this.csvFile = parseFile(csvFile);
+        fillUi();
+    }
+    
+    private CsvFile parseFile(final File csvFile) throws IOException {
+        List<String> lines = Files.readAllLines(csvFile.toPath());
+        if (lines.size() > 1) {
```

Comment:
I think a CSV file is valid even if it only contains a header or even if it's empty.

Reply:
you're right